src/Namshi/JOSE/JWS.php

```diff
@@ -84,14 +85,16 @@ public function getTokenString()
      * @return JWS
      * @throws \InvalidArgumentException
      */
-    public static function load($jwsTokenString)
+    public static function load($jwsTokenString, Encoder $encoder = null)
     {
-        $encoder = strpbrk($jwsTokenString, '+/=') ? new Base64Encoder() : new Base64UrlSafeEncoder();
+        if ($encoder === null) {
+            $encoder = strpbrk($jwsTokenString, '+/=') ? new Base64Encoder() : new Base64UrlSafeEncoder();
+        }
         $parts   = explode('.', $jwsTokenString);
 
         if (count($parts) === 3) {
-            $header  = json_decode($encoder->decode($parts[0]), true);
-            $payload = json_decode($encoder->decode($parts[1]), true);
+            $header  = json_decode($pwet = $encoder->decode($parts[0]), true);
+            $payload = json_decode($meuh = $encoder->decode($parts[1]), true);
```

Comment:
dont worry :)

On Tue, Feb 17, 2015 at 3:32 PM, Grégoire Paris <notifications@github.com>
wrote:

> In src/Namshi/JOSE/JWS.php
> <https://github.com/namshi/jose/pull/23#discussion_r24810271>:
>
> >          $parts   = explode('.', $jwsTokenString);
> >
> >          if (count($parts) === 3) {
> > -            $header  = json_decode($encoder->decode($parts[0]), true);
> > -            $payload = json_decode($encoder->decode($parts[1]), true);
> > +            $header  = json_decode($pwet = $encoder->decode($parts[0]), true);
> > +            $payload = json_decode($meuh = $encoder->decode($parts[1]), true);
>
> push -f from work then from home (or the other way around explains it).
> Sorry for that, I'll fix it rightaway!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/namshi/jose/pull/23/files#r24810271>.
>



-- 
Nadalin Alessandro
http://www.linkedin.com/in/alessandronadalin
www.odino.org
www.twitter.com/_odino_